go/grpcweb/wrapper.go

```diff
@@ -148,14 +149,17 @@ func (w *WrappedGrpcServer) handleWebSocket(wsConn *websocket.Conn, req *http.Re
 		return
 	}
 
+	ctx, cancelFunc := context.WithCancel(req.Context())
+	defer cancelFunc()
```

Comment:
Do we need to explicitly cancel this? Can we not rely on the built-in request context being cancelled?

Reply:
If we can rely on the request context we don't need to pass around a cancelFunc either. I assume you've tested some of this?

Reply:
Yeah, we've got a websocket client that makes a request every second, so the memory leak is really obvious, and I put in some logging for testing (tracking the start and end of handleWebSocket, as well as before and after the cancellation step in the reader).

I tested a bunch of different configurations. Not cancelling the request context also prevented connections from closing properly; I'm not sure why (you would expect `w.server.ServeHTTP` to pick up from the error returned from the `Body` that it would need to cancel the request, but this doesn't seem to happen).

I'll admit that I didn't go diving into the lower-level code to see what was going on - this was mostly by trial-and-error.

Reply:
OK - I'm not worried that this will cause any real problems, since the request is over once this function has exited, just thought it was a bit overzealous, but if your testing implies the necessity, I'm OK with it!